core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -1221,6 +1221,9 @@ public final void addUser(String userName, Object credential, String[] roleList,
 		try {
 			roleList = UserCoreUtil
 					.combine(doGetInternalRoleListOfUser(userName, "*"), Arrays.asList(roleList));
+            		// If the newly created user has internal roles assigned from the UI wizard those internal roles
+            		// will be duplicated in the roles list. Duplcated roles are eliminated here.
+            		roleList = new HashSet<String>(Arrays.asList(roleList)).toArray(new String[0]);
```

Comment:
There is a warning from IDE where method call toArray() with zero length argument. Please fix correctly.